limiter/limiter.go

```diff
@@ -24,11 +25,15 @@ func New(maxRequests, rps int, maxDuration time.Duration) *Run {
 	}
 
 	r := &Run{
+		unlimited:    *atomic.NewBool(maxRequests == 0),
```

Comment:
The Go memory model is stricter than other memory models,
https://golang.org/ref/mem

The tldr is
```
Programs that modify data being simultaneously accessed by multiple goroutines must serialize such access.
```

In x86, this could be a volatile, and the stale read would be OK. Go doesn't allow that though.

The data race detector has no false positives in Go, so the fact that it gave a race when I made it non-atomic means it's definitely a data race.